guides/cheat-sheets/voice.cheatmd

```diff
@@ -0,0 +1,173 @@
+# Voice Usage
+
+This cheat sheet covers basic use of the Discord Voice API through the `Nostrum.Voice` module.
+
+## Playing Audio
+{: .col-2}
+
+### Playing immediately with a `try_play` function
+
+```elixir
+def try_play(guild_id, url, type, opts \\ []) do
+  case Nostrum.Voice.play(guild_id, url, type, opts) do
+    {:error, _msg} ->
+      # Wait for handshaking to complete
+      Process.sleep(100)
+      try_play(guild_id, url, type, opts)
+
+    _ ->
+      :ok
+  end
+end
+```
+
+```elixir
+Nostrum.Voice.join_channel(guild_id, channel_id)
+try_play(guild_id, "./song.mp3", :url)
+```
+
+### Playing immediately using `Nostrum.Voice.ready?/1`
+
+```elixir
+def play_when_ready(guild_id, url, type, opts \\ []) do
+  if Nostrum.Voice.ready?(guild_id) do
+    Nostrum.Voice.play(guild_id, url, type, opts)
+  else
+    # Wait for handshaking to complete
+    Process.sleep(25)
+    play_when_ready(guild_id, url, type, opts)
+  end
+end
+```
+
+```elixir
+Nostrum.Voice.join_channel(guild_id, channel_id)
+play_when_ready(guild_id, "https://youtu.be/b4RJ-QGOtw4", :ytdl)
+```
+
+### Playing immediately by using events
+
+```elixir
+alias Nostrum.Struct.Event.VoiceReady
+alias Nostrum.Voice
+```
+
+In your consumer module
+
+```elixir
+def handle_event({:VOICE_READY, %VoiceReady{guild_id: guild_id} = _event, _v_ws_state}) do
+  Voice.play(guild_id, "~/loud_noise.mp3", :url, volume: 10)
+end
+```
+
+Once voice connection handshaking is complete, audio will begin playing
+without the need for polling functions like `try_play/4` or `play_when_ready/4`
+as shown in the other examples
+
+```elixir
+Voice.join_channel(guild_id, channel_id)
+# Playback will start automatically when ready
+```
+
+## Audio FFmpeg options
+{: .col-2}
+
+### Volume
+
+```elixir
+alias Nostrum.Voice
+```
+
+Half volume
+
+```elixir
+Voice.play(guild_id, "~/final_mix.wav", :url, 
+  volume: 0.5
+)
+```
+
+Extreme clipping
+
+```elixir
+Voice.play(guild_id, "~/boost_this.m4a", :url, 
+  volume: 100
+)
+```
+
+Normal volume with inverted phase
+
+```elixir
+Voice.play(guild_id, "~/music.mp3", :url, 
+  volume: -1.0
+)
+```
+
+### Start position and duration
+
+```elixir
+alias Nostrum.Voice
+```
+
+Start at 37.8 seconds
+
+```elixir
+Voice.play(guild_id, "https://youtu.be/b4RJ-QGOtw4", :ytdl, 
+  start_pos: "0:37.8"
+)
+```
+
+Play the first at 15.3 seconds from the beginning
+
+```elixir
+Voice.play(guild_id, "https://youtu.be/b4RJ-QGOtw4", :ytdl, 
+  duration: "15.3"
+)
+```
+
+Play for 80 seconds (equivalently 1:20) starting from 90 seconds (equivalently 1:30) in
+
+```elixir
+Voice.play(guild_id, "~/music.mp3", :url, 
+  start_pos: "90", 
+  duration: "80"
+)
+```
+
+### Advanced audio filtering
```

Comment:
For these (and maybe other places in the document), I'd look at the h4 styling that cheatmd allows, here's something I cooked up locally:

![image](https://github.com/Kraigie/nostrum/assets/20439493/0684cb6b-95e1-4429-b804-428a001bb332)

I also think that since you showcase tempo modulation in example 2, just show the cut-off filters in example 1.

Reply:
The purpose of showing samples rate coefficients of greater than or less than 1 is that one requires realtime to be set to false

Reply:
Goootcha okay, yeah of course, that makes sense. In that case, I'd maybe split this out into another example titled "Tempo modulation" or something, then show the case where you need to modify realtime and the case where you don't.